sourcetracker/_cli/gibbs.py

```diff
@@ -134,96 +131,69 @@ def gibbs(table_fp, mapping_fp, output_dir, loo, jobs, alpha1, alpha2, beta,
     # failed if so.
     os.mkdir(output_dir)
 
-    # Load the mapping file and biom table and remove samples which are not
-    # shared.
-    o = open(mapping_fp, 'U')
-    sample_metadata_lines = o.readlines()
-    o.close()
+    # Load the metadata file and feature table. Do a data check on the
+    # feature_table.
+    sample_metadata = pd.read_table(mapping_fp, index_col=0)
```

Comment:
The Qiita devs experienced some problematic behavior when loading sample metadata this way. You can work around a lot of this as illustrated [here](https://github.com/qiime2/qiime2/blob/62f878b6bddad943a0651e1d51fdc2a00007803e/qiime/metadata.py#L25).